astropy/io/fits/hdu/compressed/_tiled_compression.py

```diff
@@ -256,8 +256,7 @@ def _get_compression_setting(header, name, default):
 
 def _column_dtype(compressed_coldefs, column_name):
     tform = compressed_coldefs[column_name].format
-    if tform.startswith("1"):
-        tform = tform[1:]
+    tform = tform.removeprefix("1")
```

Comment:
At the very least
```patch
-    tform = compressed_coldefs[column_name].format
-    if tform.startswith("1"):
-        tform = tform[1:]
+    tform = compressed_coldefs[column_name].format.removeprefix("1")
```
That being said, it's not clear to me why this function prefers to write out a long `if-elif-elif` chain instead of doing a simple dictionary lookup.

Reply:
Looks like it was introduced in https://github.com/astropy/astropy/pull/14430 , performance work by @Cadair and @astrofrog